File: MoneyFlowAPI/Application/DTOs/DTO_TransactionsPage.cs
namespace MoneyFlowAPI.Application.DTOs
{
    public class DTO_TransactionsPage
    {
        public List<DTO_Transactions> Transactions { get; set; } = new();

        public decimal CurrentBalance { get; set; }

        public decimal MonthlyIncome { get; set; }

        public decimal MonthlyExpense { get; set; }

        public decimal NetFlow { get; set; }

        public int TotalTransactionsCount { get; set; }

        public int MonthlyTransactionsCount { get; set; }
    }
}

File: MoneyFlowUI/src/app/shared/dtos/DTO_TransactionsPage.ts
import { DTO_Transactions } from './DTO_Transactions';

export class DTO_TransactionsPage {
    transactions: DTO_Transactions[] = [];
    currentBalance = 0;
    monthlyIncome = 0;
    monthlyExpense = 0;
    netFlow = 0;
    totalTransactionsCount = 0;
    monthlyTransactionsCount = 0;
}
